driver/restdriver.go

```diff
@@ -37,19 +52,422 @@ func (driver *RestDriver) Initialize(logger logger.LoggingClient, asyncValues ch
 	return handler.Start()
 }
 
+// Validate data that is being received as part of the http request and response
+func (driver *RestDriver) validateData(resource models.DeviceResource, reading interface{}, valueType string, contentType string) (interface{}, error) {
```

Comment:
Why are you duplicating the code here: https://github.com/edgexfoundry/device-rest-go/blob/main/driver/resthandler.go#L162
when you could just call it from HandleReadCommands
Yes it will need to be refactored to not be a receiver function so it can be called from both places.